ckanext/googleanalytics/controller.py

```diff
@@ -56,28 +56,22 @@ def _post_analytics(
                 "ea": request_obj_type + request_function,
                 "el": id,
             }
-            # overide the event category if request call is internal
-            if request.headers.get("User-Agent", '').startswith(("frontend-v2/latest", "data-explorer/next-gen")) \
-              or (request.headers.get("Request-Source", '') in ["data-explorer"]):  
-                data_dict.update({
-                    "ec":  "Frontend/Data Explorer CKAN API Request"
-                })
+             
+            '''
+              Many other internal service also using CKAN APIs eg. frontend-v2, data-subscription, dataexplorer.
+              so, filtering internal request by user-agent and headers to avoid tracking. 
+            '''
+            internal_req_headers = request.headers.get("Request-Source", '') in ["data-explorer", "ckan-internal"]
+                                           
+            internal_req_user_agent = request.headers.get("User-Agent", '') \
+                                .startswith(("frontend-v2/latest", "data-explorer/next-gen" \
+                                "ckan-datapusher/latest", "ckan-others/latest", "data-subscription/latest")) 
             
-            if request.headers.get("User-Agent", '').startswith(("ckan-datapusher/latest", "ckan-others/latest")) \
-              or (request.headers.get("Request-Source", '') in ["ckan-internal"]):
-                data_dict.update({
-                    "ec":  "CKAN API internal Request"
-                })
-                
-            if request.headers.get("User-Agent", '').startswith(("data-subscription/latest")):
-                data_dict.update({
-                    "ec":  "Data Subscription CKAN API Request"
-                })
-
-            params_dict = self._ga_prepare_parameter(
-                request_obj_type, request_function, ids)
-            data_dict.update(params_dict)
-            plugin.GoogleAnalyticsPlugin.analytics_queue.put(data_dict)
+            if not (internal_req_headers or internal_req_user_agent):  
```

Comment:
@sagargg I think you're trying to not track calls by internal services. For that, you're checking HTTP headers - "Request-Source" and "User-Agent". Instead of having 2 separate variables for each of them, you could do:

```python
is_it_internal_request = request.headers.get('Request-Source', '') in [...] or request.headers.get('User-Agent', '').startswith(...)

if not is_it_internal_request:
    # track this request...
```